boa/contracts/abi/abi_contract.py

```diff
@@ -34,7 +33,8 @@ def __init__(self, abi: dict, contract_name: str):
 
     @property
     def name(self) -> str:
-        return self._abi["name"]
+        # note: the `constructor` definition does not have a name
+        return self._abi.get("name", "")
```

Comment:
yea i don't think we should use `""` as a default here, i think we should return Optional and just handle the None case at the call sites. like we will get stuff like `""` as a key in the overloads dictionary, which seems funky.

or if we really want a name, we can use `"__init__"` (or `"constructor"` as i think you originally had).

Reply:
yeah constructor was the first workaround I thought of, because it's just clear and simple.
handling it at the call sites is fine, but it creates a lot of boiler plate code